base/dict.jl

```diff
@@ -189,6 +189,16 @@ function grow_to!{K,V}(dest::Associative{K,V}, itr, st)
     return dest
 end
 
+
+"""
+    similar(d::Dict, [element_type=Pair{K,V}])
+
+Create an empty dictionary with key type `K` and value type `V`. The second element
```

Comment:
I think it's quite clear that `similar(d)` is much shorter than the constructor; this is even more the case when the dict is passed as an argument so that its type parameters can vary. As regards the two-argument constructor, it's indeed less obvious, but since we're not documenting a generic method for `Associative` it's hard to say anything about types other than `Dict`. I wouldn't worry too much about it.